DM-164 Orchestrate application

KochTobi opened the pull request:
Adds the application logic

KochTobi committed:
Write successfull execution time for first time run.

KochTobi committed:
Add command line parser and command

KochTobi committed:
Implement logic

KochTobi committed:
Add user feedback

KochTobi committed:
Add user feedback

KochTobi committed:
Add user feedback

KochTobi committed:
Remove unused location

sven1103 commented on src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy:
```diff
@@ -0,0 +1,61 @@
+            .filter(Result::isOk)
+            .map(Result::getValue)
+            .peek(it -> log.info("\tupdating $it"))
+            .forEach(statusReporter::reportSampleStatusUpdate)
```
What is the exception handling here? I.e. when the sample status update was not successful?

KochTobi replied:
The exception handling happens in the ReporterApp

KochTobi replied:
https://github.com/qbicsoftware/sample-status-reporter/pull/8/files#diff-a6ab529c19bba792c839bd3b1916ba590cf529ac08dead45af54f26a899c2768R45

KochTobi committed:
Remove unused location

sven1103 reviewed (commented):
Question regarding error handling :)

KochTobi committed:
Add tests

sven1103 reviewed (approved):
LGTM, i have no objections

KochTobi committed:
Fix Sample construction in RealLimsQueryService

KochTobi committed:
Remove println

KochTobi committed:
Make the last update search time point optional

In case no previous update was done this should be empty.

KochTobi committed:
Update the README.md

KochTobi committed:
Ensure only successful runs where work is done are remembered

Previous ot this commit runs with the `-h` or the `-V` arguments were remembered as last update time point

KochTobi committed:
Fix tests

sven1103 commented on README.md:
```diff
@@ -1,154 +1,41 @@
+| `USER_DB_HOST`                      | The URL to the host of the user database                               |
+| `USER_DB_NAME`                      | The database name                                                      |
+| `USER_DB_USER_NAME`                 | The database user name                                                 |
+| `USER_DB_USER_PW`                   | The database user password                                             |
```
🎉 awesome

sven1103 commented on src/main/groovy/life/qbic/samplestatus/reporter/commands/ReportSinceInstant.groovy:
```diff
@@ -0,0 +1,72 @@
+    }
+  }
+
+  private Instant defaultTimePoint = updateSearchService.getLastUpdateSearchTimePoint().orElse(Instant.now().minus(1, ChronoUnit.DAYS))
```
does this work? can you guarantee that this is executed after the updateSearchService is available to the instance?

KochTobi replied:
Yes. I tested it locally and this is called from the constructor so the updateSearchService is available at that time 😄.
And as it is only called after the `updateSearchService` is instantiated this works.

KochTobi replied:
It changed it as I am unsure how the order of execution is here and whether it is always the same

KochTobi committed:
Remove the now()

If we have this the whole Optional is unnecessary, and we do not get the desired default behaviour

sven1103 reviewed (changes requested):
Only one concern with the variable instanciation, else all good

KochTobi committed:
make it a supplier

sven1103 reviewed (approved):
Great

KochTobi committed:
make it safer
